### PR TITLE
Documentation updates + remote control demo revision

### PR DESCRIPTION
**Pull request by chester-leung**
* Revise demo documentation
* Add multiclient remote control documentation
* Remove cluster remote control documentation - we can add it back when that demo is fixed
* Revise remote-control demo to have user use non-zero key encrypted data

### CODE REVIEW

File: demo/python/remote-control/client/client.py [hunk 1]
```diff
@@ -24,14 +25,14 @@ def run(channel_addr, sym_key_file, priv_key_file, cert_file):
     enclave_reference.add_key()
 
     print("Creating training matrix")
-    dtrain = xgb.DMatrix({username: HOME_DIR + "demo/data/agaricus.txt.train.enc"}, encrypted=True)
+    dtrain = xgb.DMatrix({username: HOME_DIR + "demo/python/remote-control/data/train.enc"}, encrypted=True)
```
Comment (podcastinator) [inline demo/python/remote-control/client/client.py]: Drop `encrypted=True` here and elsewhere. This is the default, plus we do not support `encrypted=False` at the moment.

File: doc/tutorials/outsourced.rst [hunk 2]
```diff
@@ -88,16 +55,19 @@ This setup will involve encrypting data on the client, transferring the data to
       python3 encrypt.py
 
 
-3. **Send encrypted data to the server**
+2. **Send encrypted data to the server**
+
+   We assume that there will be a mechanism to transfer the encrypted data to the server. For the purposes of this demo, the user can try, for example, ``scp`` to simulate this transfer.
```
Comment (podcastinator) [inline doc/tutorials/outsourced.rst]: simulate --> perform / do

**Review by podcastinator — COMMENTED** [review]
Thanks! Left some comments.

File: demo/python/basic/secure-xgboost-demo.py [hunk 3]
```diff
@@ -47,15 +48,15 @@
 booster = xgb.train(params, dtrain, num_rounds, evals=[(dtrain, "train"), (dtest, "test")])
 
 # Save model to a file
-booster.save_model(HOME_DIR + "/demo/python/basic/modelfile.model")
+booster.save_model(HOME_DIR + "demo/python/basic/modelfile.model")
```
[the remark this answers is not in the document]
Comment (chester-leung) [reply demo/python/basic/secure-xgboost-demo.py:51]: Fixed

File: demo/python/basic/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+modelfile.model
```
[the remark this answers is not in the document]
Comment (chester-leung) [reply demo/python/basic/.gitignore]: Good idea. Let's address this in a later PR once we have a better grasp of all our demos and the files in each?

File: demo/python/remote-control/client/client.py (hunk 1 above)
Comment (chester-leung) [reply demo/python/remote-control/client/client.py]: Fixed

File: doc/tutorials/outsourced.rst (hunk 2 above)
Comment (chester-leung) [reply doc/tutorials/outsourced.rst]: Done

File: demo/python/basic/.gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (podcastinator) [reply demo/python/basic/.gitignore]: I think we can just put `*.model` in `demo/.gitignore`?

[the remark this answers is not in the document]
Comment (chester-leung) [reply demo/python/basic/.gitignore]: There's also the various `.enc` files and the key.txt files. But I've added `*.model` to it.

**Review by podcastinator — APPROVED** [review]

**Pull request merged**